packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,731 @@
+    title="Redux lifecycle with Reactive Controllers"
+    alt="A vector box diagram of the Redux cycle that is split into blue-toned, green-toned, and orange-toned boxes and arrows. At the bottom is a blue box that says UI. There is a blue arrow with the blue label 'event'. That points to a blue box called 'Event Handler'. At the top boundary of the Event Handler box is an orange box labeled 'Action Dispatcher'. From this orange box is an orange arrow labeled 'Action' pointing inside another orange box labeled Store. The orage Store box has a nested green box inside of it labeled Reducer which has 3, nested, little, green squares each labeled R. On the bottom edge of the orange Store box is a red box that says 'State'. The Orange Action arrow that was pointing inside the Orange Store box is pointing to the top of a green Reducer box. Out of the bottom of the green reducer box is another orange arrow pointing to the red 'State' box, and this arrow is labeled 'Updated State'. Out of the red State box are two orange arrows. The first orange arrow is inside the orange Store box pointing back to the top of the green reducer box and is labeled 'Old State'. The other orange arrow pointing out of the 'State' box turns into a blue arrow and is pointing to the original blue 'UI' box. The orange part of this arrow is labeled 'Subscription' in orange, and the blue part of the arrow is labeled 'Reactive Controller' in blue. Everything but the orange-blue arrow is enclosed in a dashed green line that has a key that labels that section as 'Completed'">
+
+Next we need to give some elements the ability to connect and subscribe to changes to the store and update the UI. On top of that we will write a “Selector” which will select a specific datum from the overall state, and if that value changes, we will tell the Reactive Controller host to re-render.
+
+### Attaching the Controller and Accepting options
+
+First we will write the definition of the `SelectorController` class and attach it to the host element so that the Reactive Controller can hook into the host’s reactive update lifecycle:
+
+```ts
+import type { ReactiveController, ReactiveControllerHost } from 'lit';
+
+export class SelectorController implements ReactiveController {
+  host: ReactiveControllerHost;
+
+  constructor(host: ReactiveControllerHost) {
+    this.host = host;
+    host.addController(this);
+  }
+}
+```
+
+Next, let's accept the following options: the Redux store in which we would like to subscribe to, as well as the Selector we would like to use to select what data we would like to render from the store:
+
+```ts
+import type { Store, Action as RAction } from '@reduxjs/toolkit';
+...
+
+export type Selector<State, Result> = (state: State) => Result;
+
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  host: ReactiveControllerHost;
+  store: Store<State, Action>;
+  selector: Selector<State, Result>;
```

Comment:
Nit: It probably makes sense to mark all of these as private and do either `#` or `_` prefix.